src/compiler/compile.js

```diff
@@ -588,14 +590,27 @@ function checkTerminalDirectives (el, options) {
       return skip
     }
   }
-  var value, dirName
-  for (var i = 0, l = terminalDirectives.length; i < l; i++) {
-    dirName = terminalDirectives[i]
-    value = el.getAttribute('v-' + dirName)
-    if (value != null) {
-      return makeTerminalNodeLinkFn(el, dirName, value, options)
+
+  var attrs, attr, name, value, matched, dirName, def, termDef
+  attrs = el.attributes
+  for (var i = 0, j = attrs.length; i < j; i++) {
+    attr = attrs[i]
+    if ((matched = attr.name.match(dirAttrRE))) {
+      def = resolveAsset(options, 'directives', matched[1])
+      if (def && def.terminal) {
+        if (!termDef || ((def.priority || DEFAULT_TERMINAL_PRIORITY) > termDef.priority)) {
+          termDef = def
+          name = attr.name
+          value = attr.value
+          dirName = matched[1]
+        }
+      }
     }
   }
+
+  if (termDef) {
+    return makeTerminalNodeLinkFn(el, dirName, value, options, termDef, name)
```

Comment:
we can already extract arg as `matched[2]` here, and can just use that and pass it to `makeTerminalNodeLinkFn` instead of using `argRE`.